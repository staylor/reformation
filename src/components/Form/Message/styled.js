import { css } from 'pretty-lights';
import themeUtils from 'styles/theme';

export const messageWrapClass = css`
  background: ${themeUtils.colors.background};
  border-left: 4px solid ${themeUtils.colors.pink};
  box-shadow: 0 1px 1px 0 rgba(0, 0, 0, 0.1);
  display: block;
  margin: 5px 0 15px;
  padding: 1px 38px 1px 12px;
  position: relative;
`;

export const messageTextClass = css`
  color: ${themeUtils.colors.dark};
  font-size: 13px;
  line-height: 1.5;
  margin: 0.5em 0;
  padding: 2px;
`;

export const dismissButtonClass = css`
  background: none;
  border: none;
  cursor: pointer;
  margin: 0;
  padding: 9px;
  position: absolute;
  right: 1px;
  top: 0;

  &::before {
    background: none;
    color: ${themeUtils.colors.text};
    content: '\f153';
    display: block;
    /* stylelint-disable-next-line font-family-no-missing-generic-family-keyword */
    font: normal 16px/20px dashicons;
    height: 20px;
    speak: none;
    text-align: center;
    width: 20px;
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
  }
`;
