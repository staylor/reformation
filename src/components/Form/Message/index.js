import React, { Component } from 'react';
import { messageWrapClass, messageTextClass, dismissButtonClass } from './styled';

/* eslint-disable react/prop-types,jsx-a11y/control-has-associated-label */

export default class Message extends Component {
  state = {
    hidden: false,
  };

  onClick = () => {
    this.setState({ hidden: true });
  };

  render() {
    return this.state.hidden ? null : (
      <div className={messageWrapClass}>
        <p className={messageTextClass}>{this.props.text}</p>
        <button type="button" className={dismissButtonClass} onClick={this.onClick} />
      </div>
    );
  }
}
