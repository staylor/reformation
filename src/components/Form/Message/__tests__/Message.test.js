import React from 'react';
import Message from 'components/Form/Message';
import { shallow } from 'enzyme';

describe('Message', () => {
  test('empty', () => {
    const wrapper = shallow(<Message />);

    expect(wrapper).toMatchSnapshot();
  });

  test('text', () => {
    const wrapper = shallow(<Message text="This is an admin message." />);

    expect(wrapper).toMatchSnapshot();
  });
});
