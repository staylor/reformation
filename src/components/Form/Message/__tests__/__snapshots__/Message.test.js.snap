// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message empty 1`] = `
.lights-2 {
  background: #fafafa;
  border-left: 4px solid #e50082;
  box-shadow: 0 1px 1px 0 rgba(0,0,0,0.1);
  display: block;
  margin: 5px 0 15px;
  padding: 1px 38px 1px 12px;
  position: relative;
}

.lights-0 {
  color: #262626;
  font-size: 13px;
  line-height: 1.5;
  margin: 0.5em 0;
  padding: 2px;
}

.lights-1 {
  background: none;
  border: none;
  cursor: pointer;
  margin: 0;
  padding: 9px;
  position: absolute;
  right: 1px;
  top: 0;
}

.lights-1::before {
  background: none;
  color: #999;
  content: '\\f153';
  display: block;
  font: normal 16px/20px dashicons;
  height: 20px;
  speak: none;
  text-align: center;
  width: 20px;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

<div
  className="lights-2"
>
  <p
    className="lights-0"
  />
  <button
    className="lights-1"
    onClick={[Function]}
    type="button"
  />
</div>
`;

exports[`Message text 1`] = `
.lights-2 {
  background: #fafafa;
  border-left: 4px solid #e50082;
  box-shadow: 0 1px 1px 0 rgba(0,0,0,0.1);
  display: block;
  margin: 5px 0 15px;
  padding: 1px 38px 1px 12px;
  position: relative;
}

.lights-0 {
  color: #262626;
  font-size: 13px;
  line-height: 1.5;
  margin: 0.5em 0;
  padding: 2px;
}

.lights-1 {
  background: none;
  border: none;
  cursor: pointer;
  margin: 0;
  padding: 9px;
  position: absolute;
  right: 1px;
  top: 0;
}

.lights-1::before {
  background: none;
  color: #999;
  content: '\\f153';
  display: block;
  font: normal 16px/20px dashicons;
  height: 20px;
  speak: none;
  text-align: center;
  width: 20px;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

<div
  className="lights-2"
>
  <p
    className="lights-0"
  >
    This is an admin message.
  </p>
  <button
    className="lights-1"
    onClick={[Function]}
    type="button"
  />
</div>
`;
